gst/gstvalue.h

```diff
@@ -78,12 +89,13 @@ G_BEGIN_DECLS
  *
  * Can be used together with #GST_FOURCC_FORMAT to properly output a
  * #guint32 fourcc value in a printf()-style text message.
+ * For invalid fourcc substitutes non ascii chars by '?'.
  */
-#define GST_FOURCC_ARGS(fourcc) \
-        ((gchar) ((fourcc)     &0xff)), \
-        ((gchar) (((fourcc)>>8 )&0xff)), \
-        ((gchar) (((fourcc)>>16)&0xff)), \
-        ((gchar) (((fourcc)>>24)&0xff))
+#define GST_FOURCC_ARGS(fourcc) (\
+        GST_PRINTABLE_CHAR_ARGS(fourcc & 0xff), \
```

Comment:
if fourcc is an expression, such as in GST_FOURCC_ARGS ( a + b) wrong code will be generated. It needs brackets : GST_PRINTABLE_CHAR_ARGS( (fourcc) & 0xff)

Reply:
Done